src/descriptor/tr.rs

```diff
@@ -37,6 +37,7 @@ pub enum TapTree<Pk: MiniscriptKey, Ext: Extension = NoExt> {
     // are of Leafversion::default
     Leaf(Arc<Miniscript<Pk, Tap, Ext>>),
     /// A taproot leaf denoting a spending condition in terms of Simplicity
+    #[cfg(feature = "simplicity")]
     SimplicityLeaf(Arc<simplicity::Policy<Pk>>),
```

Comment:
In d5c0d65320816bfdf36411feed4bdff0708b5b12:

We can't make an enum variant dependent on a feature-gate like this because it will make adding the feature a breaking change for anyone trying to match on the enum.

Our choices are either to mark the enum `#[non_exhaustive]` (probably the easiest thing for now) or to create a dummy `simplicity` module when the `simplicity` feature is disabled, that will include a dummy `simplicity::Policy` so that this can still compile with the variant present but no Simplicity library.

Reply:
In d9b3caf3c090da97daeb709235076798abf4782e, sorry. Not sure what that other hash is.

Reply:
mark the enums `#[non_exhaustive]`